packages/gatsby-theme-carbon/src/components/ResourceCard/ResourceCard.js

```diff
@@ -13,100 +13,140 @@ import { settings } from 'carbon-components';
-      [`${prefix}--aspect-ratio--1x1`]: aspectRatio === '1:1',
-      [`${prefix}--aspect-ratio--16x9`]: aspectRatio === '16:9',
-      [`${prefix}--aspect-ratio--4x3`]: aspectRatio === '4:3',
-    });
-
-    const carbonTileclassNames = classnames([`${prefix}--tile`], {
-      [`${prefix}--tile--clickable`]: href !== undefined,
-    });
-
-    const cardContent = (
-      <>
-        {subTitle && (
-          <h5 className={`${prefix}--resource-card__subtitle`}>{subTitle}</h5>
-        )}
-        {title && (
-          <h4 className={`${prefix}--resource-card__title`}>{title}</h4>
-        )}
-        <div className={`${prefix}--resource-card__icon--img`}>{children}</div>
-        <div className={`${prefix}--resource-card__icon--action`}>
-          {actionIcon === 'launch' && !disabled ? (
-            <Launch20 aria-label="Open resource" />
-          ) : null}
-          {actionIcon === 'arrowRight' && !disabled ? (
-            <ArrowRight20 aria-label="Open resource" />
-          ) : null}
-          {actionIcon === 'download' && !disabled ? (
-            <Download20 aria-label="Download" />
-          ) : null}
-          {actionIcon === 'email' && !disabled ? (
-            <Email20 aria-label="Email" />
-          ) : null}
-          {actionIcon === 'disabled' || disabled === true ? (
-            <Error20 aria-label="disabled" />
-          ) : null}
-        </div>
-      </>
-    );
+const ResourceCard = ({
+  children,
+  href,
+  subTitle,
+  title,
+  color = 'light',
+  disabled = false,
+  aspectRatio = '2:1',
+  actionIcon = 'launch',
+  className,
+  type = 'regular',
+  ...rest
+}) => {
+  let isLink;
+  if (href !== undefined) {
+    isLink = href.charAt(0) === '/';
+  }
 
-    let cardContainer;
-    if (disabled === true || href === undefined) {
-      cardContainer = <div className={carbonTileclassNames}>{cardContent}</div>;
-    } else if (isLink === true) {
-      cardContainer = (
-        <Link to={href} className={carbonTileclassNames} {...rest}>
-          {cardContent}
-        </Link>
-      );
-    } else {
-      cardContainer = (
-        <a href={href} className={carbonTileclassNames} {...rest}>
-          {cardContent}
-        </a>
-      );
-    }
-
-    return (
-      <div className={ResourceCardClassNames}>
-        <div className={aspectRatioClassNames}>
-          <div className={`${prefix}--aspect-ratio--object`}>
-            {cardContainer}
+  const ResourceCardClassNames = classnames([`${prefix}--resource-card`], {
+    [className]: className,
+    [`${prefix}--resource-card--disabled`]: disabled,
+    [`${prefix}--resource-card--dark`]: color === 'dark',
+  });
+
+  const aspectRatioClassNames = classnames([`${prefix}--aspect-ratio`], {
+    [`${prefix}--aspect-ratio--2x1`]: aspectRatio === '2:1',
+    [`${prefix}--aspect-ratio--1x1`]: aspectRatio === '1:1',
+    [`${prefix}--aspect-ratio--16x9`]: aspectRatio === '16:9',
+    [`${prefix}--aspect-ratio--4x3`]: aspectRatio === '4:3',
+  });
```

Comment:
The arguments to `classnames()` don't have to be an array, I think we can also use a string template to clean this up and prevent a never ending list scenario.

```suggestion
	const aspectRatioClassNames = classnames(
      `${prefix}--aspect-ratio`,
      `${prefix}--aspect-ratio--${aspectRatio}`
    );
```

Maybe double check some of the other classnames function for the unnecessary arrays

Reply:
Got it. When I do this, users will need to pass `2x1` instead of `2:1`. Does that make a breaking change?

Reply:
Cancel that. Got it working with replace method.